docs/DeveloperGuide.md

```diff
@@ -1,34 +1,565 @@
 # Developer Guide
 
-## Design & implementation
+## Table of Content
+1. [Introduction](#1-introduction)
+<br/>&nbsp;1.1 [Purpose](#11-purpose)
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+2. [Setting up](#2-setting-up)
+3. [Design](#3-design)
+<br/>&nbsp;3.1 [Architecture](#31-architecture)
+<br/>&nbsp;3.2 [UI Component](#32-ui-component)
+<br/>&nbsp;3.3 [Parser Component](#33-parser-component)
+<br/>&nbsp;3.4 [Command Component](#34-command-component)
+<br/>&nbsp;3.5 [AnimeData Component](#35-animedata-component)
+<br/>&nbsp;3.6 [User Component](#36-user-component)
+<br/>&nbsp;3.7 [Storage Component](#37-storage-component)
 
+4. [Implementation](#4-implementation)
+<br/>&nbsp;4.1 [Estimation Feature](#41-estimation-feature)
+<br/>&nbsp;4.2 [Browse Feature](#42-browse-feature)
+<br/>&nbsp;4.3 [Workspace Feature](#43-workspace-feature)
+<br/>&nbsp;4.4 [Watchlist Management Feature](#44-watchlist-management-feature)
 
-## Product scope
-### Target user profile
+5.  [Documentation, Logging, Testing, and DevOps](#5-documentation-logging-testing-and-devops)
+<br/>&nbsp;5.1 [Documentation](#51-documentation)
+<br/>&nbsp;5.2 [Logging](#52-logging)
+<br/>&nbsp;5.3 [Testing](#53-testing)
+<br/>&nbsp;5.4 [DevOps](#54-devops)
 
-{Describe the target user profile}
+<br/>
 
-### Value proposition
+## 1. Introduction
+**AniChan** is a command-line application written in **Java 11**. It is written using the Object-Oriented Programming (OOP) paradigm which provides us with means to structure a software program into organized, reusable and reusable pieces of code that makes it good for future improvements and revisions.
+<br/>
```

Comment:
Should a more detailed introduction of the program be included here to introduce the functionality of the application?